robot/impl/local_robot_test.go

```diff
@@ -4443,3 +4443,72 @@ func TestRemovingOfflineRemotes(t *testing.T) {
 	cancelReconfig()
 	wg.Wait()
 }
+
+// Tests that machine state properly reports initializing and running.
+//func TestMachineState(t *testing.T) {
```

Comment:
This test will not work, as we need to reach a robot after it's initialized but before it's completed its reconfiguration with the full processed config. `setupLocalRobot` will skip over that intermediate state. I want to start the server as a separate process and delay the completion of its reconfiguration with the full processed config. Still thinking about how to do that.